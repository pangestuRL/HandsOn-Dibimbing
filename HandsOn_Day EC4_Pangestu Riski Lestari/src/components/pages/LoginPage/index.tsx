import React, { useState } from 'react';
import Button from '@/components/UI/Button';
import Input from '@/components/UI/Input';

const LoginPage = () => {
  const [username, setusername] = useState('');
  const [password, setpassword] = useState('');

  return (
    <>
      <>
        <label htmlFor="username">username</label>
        <Input
          onChange={(e: any) => setusername(e.target.value)}
          type="text"
          id="username"
        />
      </>
      <>
        <label htmlFor="password">password</label>
        <Input
          onChange={(e: any) => setpassword(e.target.value)}
          type="password"
          id="password"
        />
      </>
      <Button
        onClick={() =>
          console.log(`username: ${username}, password: ${password}`)
        }
      >
        login
      </Button>
    </>
  );
};

export default LoginPage;
