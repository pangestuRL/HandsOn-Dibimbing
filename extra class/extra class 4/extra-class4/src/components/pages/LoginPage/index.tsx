import React, { useState } from 'reat';
import Button from '@/component/UI/Button';
import Input from '@/components/UI/input';

const LoginPage = () => {
    const [username, setUsername] = useState('');
    const [password, setPassword] = useState('');

    return(
        <>
            <>
                <label htmlFor="username">username</label>
                <Input
                    onChange={(e:any) => setUsername(e.target.value)}
                    type="text"
                    id="username"
                />
            </>
            <>
                <label htmlFor="password">password</label>
                <input
                    onChange={(e:any) => setPassword(e.target.value)}
                    type="password"
                    id="password"
                />
            </>
            <Button
                onClick={() => 
                    console.log(`username: ${username}, password: ${password}`)
                }
            >Login
            </Button>
        </>
    );
};

export default LoginPage;