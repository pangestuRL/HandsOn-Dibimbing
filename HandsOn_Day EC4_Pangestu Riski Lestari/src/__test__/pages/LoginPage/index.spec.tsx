import LoginPage from '@/components/pages/LoginPage';
import { fireEvent, render } from '@testing-library/react';

describe('LoginPage component test', () => {
  it('should render', () => {
    const { container } = render(<LoginPage />);

    expect(container).toMatchSnapshot();
  });

  it('should check if label username available', () => {
    const { getByLabelText } = render(<LoginPage />);

    expect(getByLabelText('username')).toBeInTheDocument();
  });

  it('should click the button', () => {
    const { getByText } = render(<LoginPage />);

    const button = getByText('login');
    fireEvent.click(button);

    expect(button).toBeInTheDocument();
  });

  it('should handle login', () => {
    const { getByLabelText, getByText } = render(<LoginPage />);
    const usernameInput = getByLabelText('username');
    const passwordInput = getByLabelText('password');
    const buttonLogin = getByText('login');

    // expect render
    expect(usernameInput).toBeInTheDocument();
    expect(passwordInput).toBeInTheDocument();
    expect(buttonLogin).toBeInTheDocument();

    fireEvent.change(usernameInput, { target: { value: 'kyra@email.com' } });
    fireEvent.change(passwordInput, { target: { value: '123456' } });
    fireEvent.click(buttonLogin);

    // expect value
    expect(usernameInput).toHaveValue('kyra@email.com');
    expect(passwordInput).toHaveValue('123456');
  });
});
