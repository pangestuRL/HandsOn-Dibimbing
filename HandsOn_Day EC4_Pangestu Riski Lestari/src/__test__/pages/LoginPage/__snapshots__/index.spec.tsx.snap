// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage component test should render 1`] = `
<div>
  <label
    for="username"
  >
    username
  </label>
  <input
    id="username"
    type="text"
  />
  <label
    for="password"
  >
    password
  </label>
  <input
    id="password"
    type="password"
  />
  <button>
    login
  </button>
</div>
`;
